doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -82,6 +82,36 @@ To coerce an object :g:`t:C t₁..tₙ` of ``C`` towards ``D``, we have to
 apply the coercion ``f`` to it; the obtained term :g:`f _.._ t` is
 then an object of ``D``.
 
+Reverse Coercions
+-----------------
+
+A term is reverse-coerced only if it cannot be coerced (directly) to its expect type.
+A coercion is said to be reversible if it is declared with with the :attr:`reverse` attribute.
+
+In order to cast a term :g:`a : A` to type :g:`B` using a reverse coercion we:
+
+1. find a reversible coercion :g:`f : B >-> A`
+
+2. synthesize  some :g:`?x : B` such that :g:`f ?x = a`,
+   typically through :ref:`canonicalstructures` or :ref:`typeclasses`
+
+3. replace :g:`a` with the value of :g:`?x`
+
+If there is no such reversible coercion :g:`f : B >-> A`, then we:
+
+1. find a midpoint :g:`C` equipped with an incoming reversible coercion :g:`g : B >-> C` and
+   a coercion :g:`h : A >-> C` (not necessarily reversible)
+ 
+2. synthesize  some :g:`?x : B` such that :g:`g ?x = h a`
+
+3. replace :g:`a` with the value of :g:`?x`
+
+If more than one midpoint :g:`C` exists, we try the procedure with all of them in the following order:
+a midpoint :g:`C` is tried before a midpoint :g:`D` if there exists a coercion from :g:`C` to :g:`D`.
+The intuition is that since coercions forget information, :g:`D` has less information that :g:`C`, and hence
+inferring :g:`?x : B` from :g:`h a : D` would be harder.
```

Comment:
```suggestion
If there's more than one midpoint, such as :g:`C` and :g:`D`, Coq tries :g:`C` before :g`D` if
there is a coercion from :g:`C` to :g:`D`.
The intuition is that since coercions forget information, :g:`D` has less information that :g:`C`, and hence
inferring :g:`?x : B` from :g:`h a : D` would be harder.
```

If this is on a graph, then the order isn't entirely specified, is it?  I think needs a bit more clarity on that.
Make sure I've not misstated the behavior.

Reply:
Right, I should mention that this is well defined only if the graph is a semilattice CC @CohenCyril